valis/slide_io.py

```
@@ -1693,11 +1693,7 @@ def create_metadata(self):
             slide_meta.channel_names = None
 
         if self.is_ome:
-            toilet_roll = pyvips.Image.new_from_file(self.src_f, n=-1, subifd=-1)
```

What is the meaning of this logic? I get that its sort of a brute force way to to measure the number of channels, but why not just look at the OME metadata which says clearly in it the number of channels?

If the OME metadata is wrong or does not have the expected element then we can throw an error, but I feel like number of channels is something that should always be in the XML document

---

I was using this approach to determine the number of channels in the event that the OME metadata is incorrect, which seems to be a somewhat common occurrence, especially for non-RGB images. Since libvips isn't actually reading the image, but only reading metadata, this is a small calculation, and is forgiving of user error. I guess I would prefer to do it this way, instead of throwing an error down the line and requiring that the user resave their image with the correct OME metadata.

---

I read up on the TIFF spec a bit and I get it now. This was causing me problems because I constructed a pyramidal tiff where the resolution series was in IFDs and not in subIFDs (I saved my image with `pyramidal=True` _without_ passing `subifd=True`). 

Roger on pyvips not slurping, agree this is fast. 

Digging into the OME-TIFF spec a bit more I see now that the number of channels can be specified multiple times (across multiple `Image` elements), and also multiple times within each Image element (across multiple `<Pixel>` elements). Anyway, I am now appreciating this approach haha.